mlflow/metrics/genai/genai_metric.py

```diff
@@ -558,4 +604,104 @@ def eval_fn(
         version=version,
         metric_details=evaluation_context["eval_prompt"].__str__(),
         metric_metadata=metric_metadata,
+        genai_metric_args=genai_metric_args,
     )
+
+
+def _filter_by_field(df, field_name, value):
+    return df[df[field_name] == value]
+
+
+def _deserialize_genai_metric_args(args_dict):
+    mlflow_version_at_ser = args_dict.get("mlflow_version")
+    fn_name = args_dict.get("fn_name")
+    if fn_name is None or mlflow_version_at_ser is None:
+        raise MlflowException(
+            message="The artifact JSON file appears to be corrupted and cannot be deserialized. "
+            "Please regenerate the custom metrics and rerun the evaluation. "
+            "Ensure that the file is correctly formatted and not tampered with.",
+            error_code=PUBLIC_UNDOCUMENTED,
```

Comment:
```suggestion
            error_code=INTERNAL_ERROR,
```

Reply:
oh, I thought this error is user facing; when do we use public_undocumented?